pkg/kubectl/cmd/apply.go

```diff
@@ -240,6 +240,18 @@ func RunApply(f cmdutil.Factory, cmd *cobra.Command, out, errOut io.Writer, opti
 			visitedNamespaces.Insert(info.Namespace)
 		}
 
+		// Add change-cause annotation to resource info if it should be recorded
+		if cmdutil.ShouldRecord(cmd, info) {
+			recordInObj := info.VersionedObject
+			if info.VersionedObject != nil {
+				recordInObj = info.Object
+			}
+			if err := cmdutil.RecordChangeCause(recordInObj, f.Command(cmd, false)); err != nil {
```

Comment:
`RecordChangeCause` changes annotations of given object (for creating that object later); `ChangeResourcePatch` does a similar change but returns a patch instead (for patching that object later).